acme/client.go

```diff
@@ -432,13 +434,9 @@ func (c *Client) chooseSolvers(auth authorization, domain string) map[int]solver
 				logf("[INFO][%s] acme: Could not find solver for: %s", domain, auth.Challenges[idx].Type)
 			}
 		}
-
-		// If we can solve the whole combination, return the solvers
-		if len(solvers) == len(combination) {
-			return solvers
-		}
 	}
-	return nil
+
+	return solvers
```

Comment:
The check that was deleted on line 437 is essential tho. If this check fails we can't solve the whole combination and should not try to do so. A combination can contain multiple challenges for one domain.

It is true that we should probably try more then one solver if it fails.